etsin_finder/frontend/js/components/qvain/files/fileSelector.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { inject, observer } from 'mobx-react'
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faFolder, faChevronRight, faChevronDown } from '@fortawesome/free-solid-svg-icons'
+import { Checkbox } from '../general/form'
+import {
+  FileIcon,
+} from '../general/buttons'
+import { List, ListItem } from '../general/list'
+
+export class FileSelectorBase extends Component {
```

Comment:
Component to visualize the file and directory hierarchy to the user and provide a way to select directories and files.